app/population/middleware/set-population.js

```diff
@@ -0,0 +1,58 @@
+const locationsFreeSpacesService = require('../../../common/services/locations-free-spaces')
+const moveService = require('../../../common/services/move')
+const populationService = require('../../../common/services/population')
+
+async function setPopulation(req, res, next) {
+  try {
+    const { locationId, date } = req.params
+    let transfersIn = 0
+    let transfersOut = 0
+
+    const dailyFreeSpace = await locationsFreeSpacesService.getPrisonFreeSpaces(
+      {
+        dateFrom: date,
+        dateTo: date,
+        locationIds: locationId,
+      }
+    )
+
+    const freeSpacePopulation =
+      dailyFreeSpace?.[0]?.meta?.populations?.[0] || {}
+    const { id: populationId } = freeSpacePopulation
+
+    if (populationId) {
+      req.population = await populationService.getById(populationId, {
+        include: ['moves_from', 'moves_to', 'location'],
```

Comment:
So, previously we've tried to avoid defining includes within middleware/controllers and to keep these hidden within the service layer (which is responsible for interacting with the API).

I think it would be safer, and cleaner to set that include within `populationService.getById` rather than in this middleware.

Reply:
Moved this into a `getByIdWithMoves` method.